### PR TITLE
CNDE-1887: Update for INACTIVE Lab Reports

### PR DESCRIPTION
**Pull request by upasanapattnaik-eq**
## Notes

- Update record_status_cd for Results for deleted Lab Reports: The record is marked inactive for the Order and all associated observations. 
- Remove deleted Results from LAB_TEST and downstream tables. 

## JIRA

- **Related story**: [Jira Ticket](https://cdc-nbs.atlassian.net/browse/CNDE-1887?atlOrigin=eyJpIjoiOGEwM2Y2NGFmMzI4NDIyZTljZjVkOTJlNTJiOWRiMjQiLCJwIjoiaiJ9)

## Checklist

- [x] PR focuses on a single story.
- [ ] New unit tests added and ensured they pass.
- [ ] Service has been tested in local and it works as expected.
- [ ] Documentation has been updated for this code change (if needed).
- [ ] Code follows the Java Coding Conventions (https://www.oracle.com/java/technologies/javase/codeconventions-programmingpractices.html).

## Types of changes

What types of changes does this PR introduces?

- [ ] Bugfix
- [ ] New feature
- [ ] Breaking change

## Testing

- [ ] Does this PR has >90% code coverage?
- [ ] Is the screenshot attached for code coverage?
- [ ] Does the `gradle build` pass in your local? 
- [ ] Is the `gradle build` logs attached?

### CODE REVIEW

File: liquibase-service/src/main/resources/db/rdb_modern/routines/018-sp_d_lab_test_postprocessing-001.sql [hunk 1]
```diff
@@ -3063,6 +3051,49 @@ BEGIN
...
+        FROM
+            dbo.LAB_TEST lt
+        WHERE
+            root_ordered_test_pntr IN
+            (
+                SELECT l.root_ordered_test_pntr
+                FROM dbo.LAB_TEST l
+                WHERE l.lab_test_type = 'Order'
+                  AND l.record_status_cd = 'INACTIVE')
+          AND record_status_cd <> 'INACTIVE';
```
Comment (joshua-mills-cdw) [inline liquibase-service/src/main/resources/db/rdb_modern/routines/018-sp_d_lab_test_postprocessing-001.sql:3074]: 👍

File: liquibase-service/src/main/resources/db/rdb_modern/routines/018-sp_d_lab_test_postprocessing-001.sql [hunk 2]
```diff
@@ -3085,22 +3086,26 @@ BEGIN
...
+
+                    select lt.LAB_TEST_UID,
+                           lt.PARENT_TEST_PNTR,
+                           lt.ROOT_ORDERED_TEST_PNTR
+                    from dbo.LAB_TEST lt
+                             inner join removed_obs o on lt.PARENT_TEST_PNTR = o.LAB_TEST_UID
+                )
+                DELETE FROM dbo.LAB_TEST WHERE LAB_TEST_UID IN (SELECT LAB_TEST_UID FROM removed_obs);
+            END
+
```
Comment (joshua-mills-cdw) [inline liquibase-service/src/main/resources/db/rdb_modern/routines/018-sp_d_lab_test_postprocessing-001.sql:3108]: 👍 , looks good, was the recursive CTE just not terminating when there weren't any records with ROOT_ORDERED_TEST_PNTR = 1?

Comment (upasanapattnaik-eq) [reply liquibase-service/src/main/resources/db/rdb_modern/routines/018-sp_d_lab_test_postprocessing-001.sql:3108]: Yep. It is better to run it if the condition exists.

**Review by joshua-mills-cdw — APPROVED** [review]
LGTM!

**Pull request merged**